indico/modules/events/abstracts/forms.py

```diff
@@ -113,3 +123,53 @@ class CreateEmailTemplateForm(EditEmailTemplateRuleForm):
         ('reject', _('Reject')),
         ('merge', _('Merge'))
     ], description=_("The template that will be used as a basis for this notification. You can customize it later."))
+
+
+class AbstractForm(IndicoForm):
+    title = StringField(_("Title"), [DataRequired()])
+    description = IndicoMarkdownField(_('Content'), [DataRequired()], editor=True, mathjax=True)
+    submitted_contrib_type = QuerySelectField(_("Type"), get_label='name', allow_blank=True,
+                                              blank_text=_("No type selected"))
+    person_links = AbstractPersonLinkListField(_("People"), [DataRequired()])
+    submission_comment = TextAreaField(_("Comments"))
+    attachments = HiddenField(_('Attachments'), widget=DropzoneWidget(style='thin'))
+
+    def __init__(self, *args, **kwargs):
+        self.abstract = kwargs.pop('abstract', None)
+        super(AbstractForm, self).__init__(*args, **kwargs)
+        self.submitted_contrib_type.query = self.event.contribution_types
+        if not self.submitted_contrib_type.query.count():
+            del self.submitted_contrib_type
+        self.submitted_for_tracks.query = Track.query.with_parent(self.event).order_by(Track.title)
+        if abstracts_settings.get(self.event, 'allow_attachments'):
+            tpl = get_template_module('forms/_dropzone_themes.html')
+            self.attachments.widget.options['previewTemplate'] = tpl.thin_preview_template()
+            self.attachments.widget.options['dictRemoveFile'] = tpl.remove_icon()
+        else:
+            del self.attachments
+        self.person_links.require_speaker_author = abstracts_settings.get(self.event, 'speakers_required')
+        self.person_links.allow_speakers = abstracts_settings.get(self.event, 'allow_speakers')
+
+
+class SingleTrackMixin(object):
+    submitted_for_tracks = QuerySelectField(_("Tracks"), get_label=lambda x: x.title)
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        if abstracts_settings.get(self.event, 'tracks_required'):
+            inject_validators(self, 'submitted_for_tracks', [DataRequired()])
+        super(SingleTrackMixin, self).__init__(*args, **kwargs)
+        if not abstracts_settings.get(self.event, 'tracks_required'):
+            self.submitted_for_tracks.allow_blank = True
```

Comment:
What about always setting this and just using different blank texts ("No track selected" / "Select a track")? That way people have to select a track explicitly instead of just having the first one preselected.